src/odemis/driver/keysight.py

```diff
@@ -87,75 +105,111 @@ def _sendCommand(self, cmd):
         """
         c (int): the channel to set the low voltage. 1 or 2
-        v (int): the voltage value
+        v (float): the low voltage value
         """
-        self._sendCommand("sour%d:volt:low %d" % (c, v))
+        self._sendCommand(b"sour%d:volt:low %d" % (c, v))
 
     def setVoltageMax(self, c, v):
         """
-        c (int): the channel to set the period of. 1 or 2
-        p (str): the period value
+        c (int, 1 or 2): the channel to set the high voltage of.
+        v (float): the high voltage value
         """
-        self._sendCommand("sour%d:volt:high %d" % (c, v))
+        self._sendCommand(b"sour%d:volt:high %d" % (c, v))
 
     def setVoltageLimitMin(self, c, v):
         """
-        c (int): the channel to set the period of. 1 or 2
-        p (str): the period value
+        c (int, 1 or 2): the channel to set the minimum voltage limit of.
+        v (float): the minimum voltage limit
         """
-        self._sendCommand("sour%d:volt:lim:low %d" % (c, v))
+        self._sendCommand(b"sour%d:volt:lim:low %d" % (c, v))
 
     def setVoltageLimitMax(self, c, v):
         """
-        c (int): the channel to set the period of. 1 or 2
-        p (str): the period value
+        c (int, 1 or 2): the channel to set the maximum voltage limit of.
+        v (float): the maximum voltage limit
         """
-        self._sendCommand("sour%d:volt:lim:high %d" % (c, v))
+        self._sendCommand(b"sour%d:volt:lim:high %d" % (c, v))
+
+    def setDelay(self, c, d):
+        """
+        c (int, 1 or 2): the channel to set the delay of.
+        d (float): the trigger delay to set. between 0 and 1000 s
+        """
+        self._sendCommand(b"trig%d:del %f" % (c, d))
 
     def setDutyCycle(self, c, dc):
         """
-        c (int): the channel to set the period of. 1 or 2
-        p (str): the period value
+        c (int, 1 or 2): the channel to set the duty cycle of.
+        dc (float): the duty cycle percentage. between 00.01 and 99.99
         """
-        self._sendCommand("sour%d:func:squ:dcyc %s" % (c, dc))
+        self._sendCommand(b"sour%d:func:squ:dcyc %f" % (c, dc))
 
     def setTracking(self, c, t):
         """
-        c (int): the channel to set the period of. 1 or 2
-        p (str): the period value
+        c (int, 1 or 2): the channel to set the tracking mode of.
+        t (str, ON, OFF or INV): the tracking mode
+        """
+        self._sendCommand(b"sour%d:trac %s" % (c, t))
+
+    def setWaveform(self, c, f):
+        """
+        c (int, 1 or 2): the channel to set the tracking mode of.
+        f (str SIN, SQU, RAMP or PULS): the wave form to generate
         """
-        self._sendCommand("sour%d:trac %f" % (c, t))
+        self._sendCommand(b"sour%d:appl:%s" % (c, f))
+
+    def _set_period(self, p: float) -> float:
+        f = 1/p
+        self.setFrequency(self._channel, f)
+        return p
+
+    def _set_delay(self, d):
+        self.setDelay(self._channel, d)
```

Comment:
What we call "delay" is what is called "phase" in the Keysight AWG. See the specification: "[The delay needs] to be converted to phase by as a ratio of 360 compared to  period. Note: because the channels are locked, the waveforms of both channels are always starting at the same time. The phase only changes the delay between the channels output and the “sync out” signal."